lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -794,19 +794,14 @@ instance
     ) => KnownAddresses (SeqState n k) where
     knownAddresses s =
         let
-            (PendingIxs ixs) =
-                pendingChangeIxs s
-            internalGap =
-                fromEnum . getAddressPoolGap . gap . internalPool $ s
-            discardUndiscoveredChange xs =
-                take (length ixs) $ drop (length xs - internalGap) xs
-            changeAddresses =
-                discardUndiscoveredChange $
-                    addresses (liftPaymentAddress @n @k) (internalPool s)
-            nonChangeAddresses =
+            -- instead of only showing as many unused change addresses as there
+            -- are pending transactions, we just show them all
+            visibleChangeAddresses = addresses (liftPaymentAddress @n @k) (internalPool s)
```

Comment:
The question is... does this break any assumptions further down the line? Is the length of `visibleNonChangeAddresses` fixed? Do we want to add info about internal vs external? Questions over questions.

Reply:
We aren't supposed to show change addresses until they've been used in a transaction... They are internal and not for users to use.

Reply:
Done